Azure Services/Azure Monitor/Agents/Migration Tools/DCR Config Generator/VMIDcrGeneratorModule/VMIDcrGeneratorModule.psm1

```diff
@@ -0,0 +1,194 @@
+
+function Find-IfVmiEnabled
+{
+    param (
+        [Parameter(Mandatory=$true)][string] $ResourceGroupName,
+        [Parameter(Mandatory=$true)][string] $WorkspaceName
+    )
+    
+    $result = Get-AzOperationalInsightsIntelligencePack -ResourceGroupName $ResourceGroupName -WorkspaceName $WorkspaceName | Where-Object {$_.name -match ".*VMInsights*" }
+    if ($result -and $result.Enabled -eq "True") {
+       $True
+    }
+    
+    return $False
+}
+
+function Get-VmiDataSources
+{
+    param (
+        [Parameter(Mandatory=$true)][bool] $ProcessAndDependencies
+    )
+    
+    $vmiPerfCounter = [ordered]@{
+        "name" = "VMInsightsPerfCounters";
+        "streams" = @("Microsoft-InsightsMetrics");
+        "samplingFrequencyInSeconds" = 60;
+        "counterSpecifiers" = @("\VmInsights\DetailedMetrics");
+    }
+    $vmiPerfCounters = @($vmiPerfCounter)
+    $vmiDataSources = [ordered]@{
+        "performanceCounters" = $vmiPerfCounters;
+    }
+    if ($ProcessAndDependencies) {
+        $vmiExtension = [ordered]@{
+        "streams" = @("Microsoft-ServiceMap");
+        "extensionName" = "DependencyAgent";
+        "extensionSettings" = @{};
+        "name" = "DependencyAgentDataSource";
+        }
+        $vmiDataSources["extensions"] = @($vmiExtension)
+    }    
+    return $vmiDataSources
+}
+
+function Get-VmiDestinations
+{
+    param (
+        [Parameter(Mandatory=$true)][string] $ResourceGroupName,
+        [Parameter(Mandatory=$true)][string] $WorkspaceName
+    )
+    
+    $workspace = Get-AzOperationalInsightsWorkspace -ResourceGroupName $ResourceGroupName -Name $WorkspaceName
+    $laDest =
+    [ordered]@{
+        "workspaceResourceId" = $workspace.ResourceId;
+        "workspaceId" = $workspace.CustomerId; 
+        "name" = "VMInsightsPerf-Logs-Dest";
+    }
+    $vmiDestinations = 
+    @{
+        "logAnalytics" = @($laDest);
+    }
+    return $vmiDestinations
+}
+
+function Get-VmiDataFlows
+{
+    param (
+        [Parameter(Mandatory=$true)][bool] $ProcessAndDependencies
+    )
+
+    $vmiDataFlows = [System.Collections.ArrayList]::new()
+    $insightDataFlow =
+    [ordered]@{
+        "streams" = @("Microsoft-InsightsMetrics");
+        "destinations" = @("VMInsightsPerf-Logs-Dest");
+    }
+    $serviceMapDataFlow =
+    [ordered]@{
+        "streams" = @("Microsoft-ServiceMap");
+        "destinations" = @("VMInsightsPerf-Logs-Dest");
+    }
+    $vmiDataFlows.Add($insightDataFlow) | Out-Null
+    if ($ProcessAndDependencies) {
+        $vmiDataFlows.Add($serviceMapDataFlow) | Out-Null
+    }
+    return $vmiDataFlows
+}
+
+function Get-VmiDcrPayload

```

Comment:
instead of using $newDataFlow two times, used different DataFlows for improviing readability.